mdal/frmts/mdal_ugrid.cpp

```diff
@@ -390,6 +473,38 @@ std::string MDAL::DriverUgrid::getTimeVariableName() const
   return "time";
 }
 
+void MDAL::DriverUgrid::parseNodeCoordinatesFrom1dMesh( const std::string &meshName, std::string &var1, std::string &var2 )
+{
+  std::vector<std::string> nodeVariablesName = MDAL::split( mNcFile->getAttrStr( meshName, "node_coordinates" ), ' ' );
+
+  if ( nodeVariablesName.size() < 2 )
+    throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Error while parsing node coordinates" );
+  else if ( nodeVariablesName.size() > 3 )
+  {
+    MDAL::Log::warning( MDAL_Status::Warn_InvalidElements, name(),
```

Comment:
add comment that format XYZ does not follow UGRID conventions and have extra vars...